components/flow-orchestration-framework/org.wso2.carbon.identity.flow.mgt/src/main/java/org/wso2/carbon/identity/flow/mgt/FlowMgtService.java

```diff
@@ -79,7 +92,32 @@ public void updateFlow(FlowDTO flowDTO, int tenantID)
      */
     public FlowDTO getFlow(String flowType, int tenantID) throws FlowMgtFrameworkException {
 
-        return flowDAO.getFlow(flowType, tenantID);
+        try {
+            OrganizationManager organizationManager = FlowMgtServiceDataHolder.getInstance().getOrganizationManager();
+            String organizationId = organizationManager.resolveOrganizationId(
+                    IdentityTenantUtil.getTenantDomain(tenantID));
+            OrgResourceResolverService orgResourceResolverService =
+                    FlowMgtServiceDataHolder.getInstance().getOrgResourceResolverService();
+            return orgResourceResolverService.getResourcesFromOrgHierarchy(
+                    organizationId,
+                    LambdaExceptionUtils.rethrowFunction(orgId -> flowRetriever(flowType, orgId)),
+                    new FirstFoundAggregationStrategy<>());
+        } catch (OrganizationManagementException | OrgResourceHierarchyTraverseException e) {
+            throw handleServerException(Constants.ErrorMessages.ERROR_CODE_GET_DEFAULT_FLOW, e,
+                    IdentityTenantUtil.getTenantDomain(tenantID));
```

Comment:
Instead of repeating the getTenantDomain call here, can we move it to a variable outside the try block and reuse it for better performance? Since this getTenantDomain isn't a simple get call and has additional conditions inside? 

Please check other places too.